crates/viewer/re_time_panel/src/data_density_graph.rs

```diff
@@ -410,6 +410,31 @@ fn smooth(density: &[f32]) -> Vec<f32> {
         .collect()
 }
 
+/// Uniformly sample events using the given sample size.
+///
+/// Each sampled event's count is reweighted to preserve the total density.
+fn uniform_sample_events(events: &[(TimeInt, u64)], sample_size: usize) -> Vec<(TimeInt, u64)> {
+    re_tracing::profile_function!();
+
+    // Use double precision to prevent rounding errors when handling many events.
+    // This ensures that we sample evenly across the entire range.
+    let step = events.len() as f64 / sample_size as f64;
+    let mut sampled = Vec::with_capacity(sample_size);
+
+    for i in 0..sample_size {
+        let idx = (i as f64 * step) as usize;
+
+        // This means we might miss the last event if rounding down, but that's acceptable.
+        if let Some(&(time, count)) = events.get(idx) {
+            // Reweight the count to preserve total density
+            let weighted_count = (count as f64 * step).round() as u64;
```

Comment:
No need to round here - just change `fn add_chunk_point` from taking an `num_events: usize` to a `weight: f32`